## Test student sur les deux tables avantage, remuneration

   #### On va utlisier test student pour savoir si les samples on a pris de les deux tables avantage, remuneration representative ou non.

  On a calculer le moyenne et  l'ecart-type sur le column 'montant_ttc' dans les deux samples: avantage, remuneration.

### 1- Poser les hypothèses :

    H0: hypothèse nulle, il n'y a pas de variation entre le populations (table basic avantage ou remuneration)
    i.e les samples on a pris ils sont representative par rapport à les table basic.
    H1: On observe une différence,les samples ils ont pas representative.

### 2- On suppose l’hypothèse H0 comme vrai.

### 3- Calculer le moyenne et l'ecart-type

  On a calculer le moyenne et  l'ecart-type sur le column 'montant_ttc' dans les deux basic tables et les deux samples: avantage, remuneration. On a les resultatles suivant:

#### 3.1 le moyenne et  l'ecart-type pour le table avantage et sa sample

avantage['avant_montant_ttc'].mean()  ---> 143.9439265440081

avantage.loc[:,'avant_montant_ttc'].std() --> 3725.823924691866

avantageSample['avant_montant_ttc'].mean() --> 141.90759729237672

avantageSample.loc[:,'avant_montant_ttc'].std()  --> 2614.0841062141385

len(avantageSample) ---> 2824322

#### 3.2 le moyenne et  l'ecart-type pour le table remuneration et sa sample

remuneration['remu_montant_ttc'].mean() -->5337.852765299928

remuneration.loc[:,'remu_montant_ttc'].std()--> 39167.18380511070

remunerationSample['remu_montant_ttc'].mean() --> 5227.276655126122

remunerationSample.loc[:,'remu_montant_ttc'].std() --> 39773.13566327895

len(remunerationSample) ---> 113224

###  4- on calcule le difference entre les moyennes

  #### On suppose le moyenne de sample est m, moyenne de basic table est M, ecart-type de la basic table est Q et le nombre de chaque sample est n, on calcul le p-value par:
   ### |m-M| / (Q/n)

#### 4.1 Calculer p-value pour le table avantage 

In [4]:
import math
M= 143.9439265440081
m= 141.90759729237672
Q =3725.823924691866 
n=2824322
pValue_avantage=abs(m-M)/(Q/math.sqrt(n))
print(pValue_avantage)

0.9185076489090103


#### Pour α =0.05, donc 1 – α = 1-0.05=0.95, on peut voir que le pValue (0.9185076489090103) est inférieur donc on accept H0. Donc le sample on a pris represent le table avantage.

#### 4.2 Calculer p-value pour le table avantage

In [5]:
import math
M= 5337.852765299928
m= 5227.276655126122
Q =39167.18380511070 
n=113224
pValue_remuniration=abs(m-M)/(Q/math.sqrt(n))
print(pValue_remuniration)

0.9499663221088217


#### Pour α =0.05, donc 1 – α = 1-0.05=0.95, on peut voir que le pValue (0.9499663221088217) est inférieur donc on accept H0. Donc le sample on a pris represent le table remuniration.